src/borg/helpers/fs.py

```diff
@@ -19,6 +21,33 @@
 py_37_plus = sys.version_info >= (3, 7)
 
 
+def ensure_dir(path, name=None, mode=None, reraise=False):
+    """
+    Ensures that the dir exists with the right permissions.
+    1) Make sure the directory exists in a race-free condition
+    2) If mode is not None and the directory has been created, give the right
+    permissions
+    3) If reraise is True, catch fatal exceptions, reraise them with a pretty
+    message and the full exception output.
+    Returns True if the file has been created and has the right permissions,
+    False otherwise
```

Comment:
Maybe we just want to use exceptions (== either not catch them or re-raise them) if something bad happened and no exception if it worked.